src/net_processing.cpp

```diff
@@ -3651,7 +3651,11 @@ void PeerManagerImpl::ProcessMessage(
 
         int64_t nTimeOffset = nTime - GetTime();
         pfrom.nTimeOffset = nTimeOffset;
-        AddTimeData(pfrom.addr, nTimeOffset);
+        if (!pfrom.IsInboundConn()) {
```

Comment:
pick also bitcoin/bitcoin#24609 which update documentation also; this 2 backports should be merged together or one-after-one

Reply:
added bitcoin#24609